src/nvim/os/fs.c

```diff
@@ -946,6 +946,31 @@ int os_mkdir_recurse(const char *const dir, int32_t mode, char **const failed_di
   return 0;
 }
 
+/// Create the parent directory of a file if it does not exist
+///
+/// @param[in] fname Full path of the file name whose parent directories
+///                  we want to create
+///
+/// @return `0` for success, libuv error code for failure.
+int os_mkdir_if_absent(char *fname)
```

Comment:
Alright, I was also wondering if it would be best to also pass the file mode as a parameter ? It's hard coded here and set to `0755`, perhaps it'd be better to actually have it as a parameter ?

Reply:
> also pass the file mode as a parameter ?

sure, a parameter to os_file_mkdir() is a good idea. But not as a parameter to `:write` , or at least that should be a followup PR/feature-request.